board/e5_mini/src/main.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (c) WiseVision 2023. All rights reserved.
+ */
+
+
+#include <inttypes.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/lora.h>
+#include <errno.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/gpio.h>
+#include <app_version.h>
+
+//adc
+#if !DT_NODE_EXISTS(DT_PATH(zephyr_user)) || \
+	!DT_NODE_HAS_PROP(DT_PATH(zephyr_user), io_channels)
+#error "No suitable devicetree overlay specified"
+#endif
+
+#define DT_SPEC_AND_COMMA(node_id, prop, idx) \
+	ADC_DT_SPEC_GET_BY_IDX(node_id, idx),
+
+/* Data of ADC io-channels specified in devicetree. */
+static const struct adc_dt_spec adc_channels[] = {
+	DT_FOREACH_PROP_ELEM(DT_PATH(zephyr_user), io_channels,
+			     DT_SPEC_AND_COMMA)
+};
+//LORA
+#define DEFAULT_RADIO_NODE DT_ALIAS(lora0)
+BUILD_ASSERT(DT_NODE_HAS_STATUS(DEFAULT_RADIO_NODE, okay),
+	     "No default LoRa radio specified in DT");
+
+#define MAX_DATA_LEN 10
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(lora_send);
+
+int mapSoilMoisture(int value, int inMin, int inMax, int outMin, int outMax) {
+    return (value - inMin) * (outMax - outMin) / (inMax - inMin) + outMin;
+}
+
+int configureRx(const struct device *const lora_dev)
+{
+	/*
+	* Frequncy Plans
+	* Europe: 863-870 Mhz
+	* United States: 902-928 Mhz
+	* Austraulia: 915-928 Mhz
+	* China 470-510 Mhz
+	* Asia/Japan 920-923 Mhz
+	* Morocco 869-870 Mhz
+	* India 865-867 Mhz
+	* Russia 864-870 Mhz
+
+	*/
+	struct lora_modem_config config;
+	config.frequency = 865100000;
+	config.bandwidth = BW_125_KHZ;
+	config.datarate = SF_10;
+	config.preamble_len = 8;
+	config.coding_rate = CR_4_5;
+	config.iq_inverted = false;
+	config.public_network = false;
+	config.tx_power = 14;
+	config.tx = false;
+
+	int ret = lora_config(lora_dev, &config);
+	if (ret < 0) {
+		LOG_ERR("LoRa config failed");
+	}
+	return ret;
+}
+
+// configure transmission
+int configureTx(const struct device *const lora_dev)
+{
+	struct lora_modem_config config;
+	config.frequency = 865100000;
+	config.bandwidth = BW_125_KHZ;
+	config.datarate = SF_10;
+	config.preamble_len = 8;
+	config.coding_rate = CR_4_5;
+	config.iq_inverted = false;
+	config.public_network = false;
+	config.tx_power = 4;
+	config.tx = true;
+
+	int ret = lora_config(lora_dev, &config);
+	if (ret < 0) {
+		LOG_ERR("LoRa config failed");
+	}
+	return ret;
+}
+
+void send_msg(const struct device *dev, int moisture){
```

Comment:
keep all fuc names in the same convention use camelCased

Reply:
https://stackoverflow.com/a/1722518

Reply:
https://www.theserverside.com/answer/Camel-case-vs-snake-case-Whats-the-difference

Reply:
I am confused. I assume, whatever we use is fine as long as we are consistent with one convention